# <center>MDS6212 Fintech Theory and Practice: Week 2 Assignment<center>

----
    
**220041006 *Yihang Li***

**Data confidential requirements:** 
*     Do not share datasets with others
*     Do not post datasets on the Web site
*     Datasets can only be used to work on assignments of MDS6212, M.Sc. in Data Science, CUHK-Shenzhen.

**Datasets:** 
*     MDS6212 Week 2 Renrendai loans.xlsx
*     MDS6212 Week 2 p2p lending platforms.xlsx

## Q1:
    1) Present two tables for the summary statistics of the key variables in Renrendai loans.xlsx and p2p lending platforms.xlsx

In [1]:
import pandas as pd
import numpy as np

In [2]:
!ls

'MDS6212 Week 2 Assignment.docx'
'MDS6212 Week 2 Renrendai loans.xlsx'
'MDS6212 Week 2 p2p lending platforms.xlsx'
'Week2 Assignment.ipynb'


In [3]:
#load two datasets
Renrendai = pd.read_excel('MDS6212 Week 2 Renrendai loans.xlsx')
P2P = pd.read_excel('MDS6212 Week 2 p2p lending platforms.xlsx')

In [4]:
Renrendai.head(1)

loanId  STATUS  BIDS  DEFAULT  AMOUNT  INTEREST  MONTHS  CREDIT  HOUSE  \
0       2  CLOSED     9        0    3000       5.0       6       7      1   

   CAR  ...  IND  CITY  PURPOSE  MARRY  AGE            opentime  \
0    1  ...   IT    北京     其他借款     未婚   33 2010-10-12 17:17:01   

            readytime title  \
0 2010-10-15 11:00:49  庆祝上线   

                                         description  nickName  
0  看着人人贷经过精心的筹备,和长时间的内测,终于正式和大家见面了.特地来发个标, 希望大家多多...        一夫  

[1 rows x 25 columns]

In [5]:
P2P.head(1)

fullname  OnlineTime_YMD  Bankrupt_WDZJ Province City  Collapse  Benign  \
0   123便利贷        20140519     20170413.0       上海   上海         1     0.0   

   Fraud  RegCapital Background  Capitaldeposit  Obtaininvest  Joinasso  \
0    0.0       500.0        民营系               0           0.0       1.0   

   Autobid  Transright  Riskdeposit  Thirdguarantee  
0        0           0          0.0             0.0

### summary statistics of the key variables in Renrendai loans.xlsx

In [7]:
Renrendai.describe().apply(
    lambda s: s.apply(lambda x: format(x, 'f')))

loanId          BIDS       DEFAULT         AMOUNT  \
count    10000.000000  10000.000000  10000.000000   10000.000000   
mean    418846.947200     24.150600      0.151300   24545.835000   
std     446432.580191     41.342608      0.358359   38280.756524   
min          2.000000      1.000000      0.000000    3000.000000   
25%      84635.250000      9.000000      0.000000    8000.000000   
50%     321945.000000     15.000000      0.000000   14400.000000   
75%     582930.500000     24.000000      0.000000   26000.000000   
max    2086049.000000    592.000000      1.000000  500000.000000   

           INTEREST        MONTHS        CREDIT         HOUSE           CAR  \
count  10000.000000  10000.000000  10000.000000  10000.000000  10000.000000   
mean      12.621900     12.237300      2.146300      0.564500      0.391700   
std        2.273689      8.091090      1.530990      0.495847      0.488155   
min        5.000000      3.000000      1.000000      0.000000      0.000000   
25%       11.000000      6.000000      1.000000      0.000000      0.000000   
50%       12.000000     12.000000      2.000000      1.000000      0.000000   
75%       13.000000     12.000000      3.000000      1.000000      1.000000   
max       24.400000     36.000000      7.000000      1.000000      1.000000   

            HOUSE_L         CAR_L    EDUCATION     WORKTIME       INCOME  \
count  10000.000000  10000.000000  9996.000000  9994.000000  9998.000000   
mean       0.228400      0.082200     2.165966     2.838003     4.309162   
std        0.419823      0.274683     0.818108     0.992755     1.335842   
min        0.000000      0.000000     1.000000     1.000000     1.000000   
25%        0.000000      0.000000     2.000000     2.000000     3.000000   
50%        0.000000      0.000000     2.000000     3.000000     4.000000   
75%        0.000000      0.000000     3.000000     4.000000     5.000000   
max        1.000000      1.000000     4.000000     4.000000     7.000000   

                AGE  
count  10000.000000  
mean      34.755500  
std        6.682708  
min       24.000000  
25%       30.000000  
50%       33.000000  
75%       38.000000  
max       53.000000

### summary statistics of the key variables in p2p lending platforms.xlsx

In [8]:
P2P.describe().apply(
    lambda s: s.apply(lambda x: format(x, 'f')))

OnlineTime_YMD    Bankrupt_WDZJ     Collapse      Benign       Fraud  \
count      1000.000000       782.000000  1000.000000  782.000000  782.000000   
mean   20148498.958000  20163303.048593     0.782000    0.098465    0.246803   
std       11351.943281     13043.220563     0.413094    0.298134    0.431427   
min    20090409.000000  20120601.000000     0.000000    0.000000    0.000000   
25%    20140917.750000  20151117.000000     1.000000    0.000000    0.000000   
50%    20150401.000000  20160801.500000     1.000000    0.000000    0.000000   
75%    20151116.250000  20171107.000000     1.000000    0.000000    0.000000   
max    20180524.000000  20190904.000000     1.000000    1.000000    1.000000   

         RegCapital Capitaldeposit Obtaininvest    Joinasso      Autobid  \
count   1000.000000    1000.000000   968.000000  968.000000  1000.000000   
mean     596.064330       0.191000     0.026860    0.054752     0.244000   
std     2328.221711       0.393286     0.161756    0.227613     0.429708   
min        2.000000       0.000000     0.000000    0.000000     0.000000   
25%      100.000000       0.000000     0.000000    0.000000     0.000000   
50%      300.000000       0.000000     0.000000    0.000000     0.000000   
75%      500.000000       0.000000     0.000000    0.000000     0.000000   
max    50000.000000       1.000000     1.000000    1.000000     1.000000   

        Transright Riskdeposit Thirdguarantee  
count  1000.000000  968.000000     968.000000  
mean      0.177000    0.021694       0.034091  
std       0.381860    0.145758       0.181557  
min       0.000000    0.000000       0.000000  
25%       0.000000    0.000000       0.000000  
50%       0.000000    0.000000       0.000000  
75%       0.000000    0.000000       0.000000  
max       1.000000    1.000000       1.000000

## Q2:
    2) Perform a logit regression and examine the relation between the default likelihood and borrower characteristics such as credit, house, car, education, work time, etc. 

### Dealing missing values

In [14]:
Renrendai.describe().columns

Index(['loanId', 'BIDS', 'DEFAULT', 'AMOUNT', 'INTEREST', 'MONTHS', 'CREDIT',
       'HOUSE', 'CAR', 'HOUSE_L', 'CAR_L', 'EDUCATION', 'WORKTIME', 'INCOME',
       'AGE'],
      dtype='object')

In [17]:
q2_Data = Renrendai[[
    'DEFAULT', 'CREDIT', 'HOUSE', 'CAR', 'EDUCATION', 'WORKTIME', 'HOUSE_L',
    'CAR_L', 'INCOME', 'AGE'
]]

In [18]:
q2_Data.describe()

DEFAULT       CREDIT         HOUSE           CAR    EDUCATION  \
count  10000.000000  10000.00000  10000.000000  10000.000000  9996.000000   
mean       0.151300      2.14630      0.564500      0.391700     2.165966   
std        0.358359      1.53099      0.495847      0.488155     0.818108   
min        0.000000      1.00000      0.000000      0.000000     1.000000   
25%        0.000000      1.00000      0.000000      0.000000     2.000000   
50%        0.000000      2.00000      1.000000      0.000000     2.000000   
75%        0.000000      3.00000      1.000000      1.000000     3.000000   
max        1.000000      7.00000      1.000000      1.000000     4.000000   

          WORKTIME       HOUSE_L         CAR_L       INCOME           AGE  
count  9994.000000  10000.000000  10000.000000  9998.000000  10000.000000  
mean      2.838003      0.228400      0.082200     4.309162     34.755500  
std       0.992755      0.419823      0.274683     1.335842      6.682708  
min       1.000000      0.000000      0.000000     1.000000     24.000000  
25%       2.000000      0.000000      0.000000     3.000000     30.000000  
50%       3.000000      0.000000      0.000000     4.000000     33.000000  
75%       4.000000      0.000000      0.000000     5.000000     38.000000  
max       4.000000      1.000000      1.000000     7.000000     53.000000

***EDUCATION, WORKTIME and INCOME have slightly degree of missing, here we just delete these missing values***

In [20]:
q2_Data = q2_Data[q2_Data['EDUCATION'].notnull()]
q2_Data.reset_index(drop = True, inplace = True)
q2_Data = q2_Data[q2_Data['WORKTIME'].notnull()]
q2_Data.reset_index(drop = True, inplace = True)
q2_Data = q2_Data[q2_Data['INCOME'].notnull()]
q2_Data.reset_index(drop = True, inplace = True)

In [21]:
q2_Data.describe()

DEFAULT       CREDIT        HOUSE          CAR    EDUCATION  \
count  9990.000000  9990.000000  9990.000000  9990.000000  9990.000000   
mean      0.151451     2.146146     0.564765     0.391892     2.166066   
std       0.358506     1.531092     0.495813     0.488197     0.818221   
min       0.000000     1.000000     0.000000     0.000000     1.000000   
25%       0.000000     1.000000     0.000000     0.000000     2.000000   
50%       0.000000     2.000000     1.000000     0.000000     2.000000   
75%       0.000000     3.000000     1.000000     1.000000     3.000000   
max       1.000000     7.000000     1.000000     1.000000     4.000000   

          WORKTIME      HOUSE_L        CAR_L       INCOME          AGE  
count  9990.000000  9990.000000  9990.000000  9990.000000  9990.000000  
mean      2.838138     0.228529     0.082282     4.308909    34.754354  
std       0.992679     0.419906     0.274808     1.335527     6.680150  
min       1.000000     0.000000     0.000000     1.000000    24.000000  
25%       2.000000     0.000000     0.000000     3.000000    30.000000  
50%       3.000000     0.000000     0.000000     4.000000    33.000000  
75%       4.000000     0.000000     0.000000     5.000000    38.000000  
max       4.000000     1.000000     1.000000     7.000000    53.000000

***Now we have 9990 observations in total***

In [23]:
# default likelihood
q2_y = q2_Data[['DEFAULT']]
# borrower characteristics
q2_X = q2_Data[[
    'CREDIT', 'HOUSE', 'CAR', 'EDUCATION', 'WORKTIME', 'HOUSE_L', 'CAR_L',
    'INCOME', 'AGE'
]]

### Perform logit regression

***Now let's perform logit regression by using statsmodels***

In [24]:
import statsmodels.api as sm

In [26]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

In [28]:
q2_X = scaler.fit(q2_X).transform(q2_X)
q2_X = sm.add_constant(q2_X)
q2_model = sm.Logit(q2_y, q2_X)
q2_result = q2_model.fit()

Optimization terminated successfully.
         Current function value: 0.330132
         Iterations 9


In [29]:
q2_result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:                DEFAULT   No. Observations:                 9990
Model:                          Logit   Df Residuals:                     9980
Method:                           MLE   Df Model:                            9
Date:                Sat, 03 Oct 2020   Pseudo R-squ.:                  0.2236
Time:                        15:44:23   Log-Likelihood:                -3298.0
converged:                       True   LL-Null:                       -4247.9
Covariance Type:            nonrobust   LLR p-value:                     0.000
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -3.1979      0.091    -35.272      0.000      -3.376      -3.020
x1            -2.8978      0.126    -23.044      0.000      -3.144      -2.651
x2             0.0713      0.036      1.968      0.049       0.000       0.142
x3            -0.2239      0.039     -5.708      0.000      -0.301      -0.147
x4            -0.3401      0.033    -10.426      0.000      -0.404      -0.276
x5             0.0090      0.034      0.264      0.792      -0.058       0.076
x6            -0.1389      0.038     -3.633      0.000      -0.214      -0.064
x7             0.0445      0.037      1.207      0.228      -0.028       0.117
x8             0.1549      0.034      4.592      0.000       0.089       0.221
x9             0.1700      0.034      4.936      0.000       0.102       0.237
==============================================================================
"""

### Rebuild Model by removing non-significant variables

***We can see from the result summary, x5('WORKTIME') has much high p-value(0.792), which means it is not significant, so as x7('CAR_L'). Thus we may rebuild our model by removing them***

In [32]:
# borrower characteristics by removing 'WORKTIME' and 'CAR_L'
q2_X = q2_Data[[
    'CREDIT', 'HOUSE', 'CAR', 'EDUCATION', 'HOUSE_L',
    'INCOME', 'AGE'
]]

In [33]:
q2_X = scaler.fit(q2_X).transform(q2_X)
q2_X = sm.add_constant(q2_X)

In [34]:
q2_model = sm.Logit(q2_y, q2_X)
q2_result = q2_model.fit()

Optimization terminated successfully.
         Current function value: 0.330206
         Iterations 9


In [35]:
q2_result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:                DEFAULT   No. Observations:                 9990
Model:                          Logit   Df Residuals:                     9982
Method:                           MLE   Df Model:                            7
Date:                Sat, 03 Oct 2020   Pseudo R-squ.:                  0.2234
Time:                        15:53:11   Log-Likelihood:                -3298.8
converged:                       True   LL-Null:                       -4247.9
Covariance Type:            nonrobust   LLR p-value:                     0.000
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -3.1951      0.091    -35.288      0.000      -3.373      -3.018
x1            -2.8951      0.126    -23.050      0.000      -3.141      -2.649
x2             0.0725      0.036      2.014      0.044       0.002       0.143
x3            -0.2053      0.036     -5.707      0.000      -0.276      -0.135
x4            -0.3400      0.033    -10.451      0.000      -0.404      -0.276
x5            -0.1364      0.038     -3.574      0.000      -0.211      -0.062
x6             0.1546      0.034      4.598      0.000       0.089       0.220
x7             0.1710      0.032      5.328      0.000       0.108       0.234
==============================================================================
"""

***Now all the variables against 'DEFAULT' are significant***

## Q3:
    3) Perform an ols regression and examine the relation between the number of bids and borrower characteristics such as credit, house, car, education, work time, etc.

### Dealing missing values

In [36]:
q3_Data = Renrendai[[
    'BIDS', 'CREDIT', 'HOUSE', 'CAR', 'EDUCATION', 'WORKTIME', 'HOUSE_L',
    'CAR_L', 'INCOME', 'AGE'
]]

In [37]:
q3_Data.describe()

BIDS       CREDIT         HOUSE           CAR    EDUCATION  \
count  10000.000000  10000.00000  10000.000000  10000.000000  9996.000000   
mean      24.150600      2.14630      0.564500      0.391700     2.165966   
std       41.342608      1.53099      0.495847      0.488155     0.818108   
min        1.000000      1.00000      0.000000      0.000000     1.000000   
25%        9.000000      1.00000      0.000000      0.000000     2.000000   
50%       15.000000      2.00000      1.000000      0.000000     2.000000   
75%       24.000000      3.00000      1.000000      1.000000     3.000000   
max      592.000000      7.00000      1.000000      1.000000     4.000000   

          WORKTIME       HOUSE_L         CAR_L       INCOME           AGE  
count  9994.000000  10000.000000  10000.000000  9998.000000  10000.000000  
mean      2.838003      0.228400      0.082200     4.309162     34.755500  
std       0.992755      0.419823      0.274683     1.335842      6.682708  
min       1.000000      0.000000      0.000000     1.000000     24.000000  
25%       2.000000      0.000000      0.000000     3.000000     30.000000  
50%       3.000000      0.000000      0.000000     4.000000     33.000000  
75%       4.000000      0.000000      0.000000     5.000000     38.000000  
max       4.000000      1.000000      1.000000     7.000000     53.000000

In [38]:
q3_Data = q3_Data[q3_Data['EDUCATION'].notnull()]
q3_Data.reset_index(drop = True, inplace = True)
q3_Data = q3_Data[q3_Data['WORKTIME'].notnull()]
q3_Data.reset_index(drop = True, inplace = True)
q3_Data = q3_Data[q3_Data['INCOME'].notnull()]
q3_Data.reset_index(drop = True, inplace = True)

In [39]:
q3_Data.describe()

BIDS       CREDIT        HOUSE          CAR    EDUCATION  \
count  9990.000000  9990.000000  9990.000000  9990.000000  9990.000000   
mean     24.116316     2.146146     0.564765     0.391892     2.166066   
std      41.242397     1.531092     0.495813     0.488197     0.818221   
min       1.000000     1.000000     0.000000     0.000000     1.000000   
25%       9.000000     1.000000     0.000000     0.000000     2.000000   
50%      15.000000     2.000000     1.000000     0.000000     2.000000   
75%      24.000000     3.000000     1.000000     1.000000     3.000000   
max     592.000000     7.000000     1.000000     1.000000     4.000000   

          WORKTIME      HOUSE_L        CAR_L       INCOME          AGE  
count  9990.000000  9990.000000  9990.000000  9990.000000  9990.000000  
mean      2.838138     0.228529     0.082282     4.308909    34.754354  
std       0.992679     0.419906     0.274808     1.335527     6.680150  
min       1.000000     0.000000     0.000000     1.000000    24.000000  
25%       2.000000     0.000000     0.000000     3.000000    30.000000  
50%       3.000000     0.000000     0.000000     4.000000    33.000000  
75%       4.000000     0.000000     0.000000     5.000000    38.000000  
max       4.000000     1.000000     1.000000     7.000000    53.000000

In [40]:
# number of bids
q3_y = q3_Data[['BIDS']]
# borrower characteristics
q3_X = q3_Data[[
    'CREDIT', 'HOUSE', 'CAR', 'EDUCATION', 'WORKTIME', 'HOUSE_L', 'CAR_L',
    'INCOME', 'AGE'
]]

### Perform OLS regression

In [41]:
q3_X = scaler.fit(q3_X).transform(q3_X)
q3_X = sm.add_constant(q3_X)

q3_model = sm.OLS(q3_y, q3_X)

q3_result = q3_model.fit()

In [42]:
q3_result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   BIDS   R-squared:                       0.173
Model:                            OLS   Adj. R-squared:                  0.172
Method:                 Least Squares   F-statistic:                     232.1
Date:                Sat, 03 Oct 2020   Prob (F-statistic):               0.00
Time:                        16:02:31   Log-Likelihood:                -50383.
No. Observations:                9990   AIC:                         1.008e+05
Df Residuals:                    9980   BIC:                         1.009e+05
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         24.1163      0.375     64.242      0.000      23.380      24.852
x1             2.8557      0.394      7.248      0.000       2.083       3.628
x2             0.7982      0.459      1.738      0.082      -0.102       1.698
x3             2.0788      0.448      4.637      0.000       1.200       2.958
x4            -1.6398      0.389     -4.218      0.000      -2.402      -0.878
x5             2.4175      0.423      5.721      0.000       1.589       3.246
x6            -2.9933      0.432     -6.924      0.000      -3.841      -2.146
x7            -1.9772      0.407     -4.854      0.000      -2.776      -1.179
x8            12.3210      0.412     29.918      0.000      11.514      13.128
x9             5.4282      0.444     12.235      0.000       4.558       6.298
==============================================================================
Omnibus:                    11602.380   Durbin-Watson:                   1.743
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          1282780.294
Skew:                           6.139   Prob(JB):                         0.00
Kurtosis:                      57.139   Cond. No.                         2.31
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Q4:
    4) Perform the Cox model (Proportional hazards model) and examine the relation between the platform default (survival) likelihood and platform characteristics such as RegCapital, Joinasso, etc.

### Cox model (Proportional hazards model)

### Dealing missing values

In [44]:
P2P.describe().columns

Index(['OnlineTime_YMD', 'Bankrupt_WDZJ', 'Collapse', 'Benign', 'Fraud',
       'RegCapital', 'Capitaldeposit', 'Obtaininvest', 'Joinasso', 'Autobid',
       'Transright', 'Riskdeposit', 'Thirdguarantee'],
      dtype='object')

In [45]:
q4_Data = P2P[['Collapse', 'RegCapital', 'Joinasso', 'Capitaldeposit', 'Obtaininvest', 'Autobid',
       'Transright', 'Riskdeposit', 'Thirdguarantee']]

In [46]:
q4_Data.describe()

Collapse    RegCapital    Joinasso  Capitaldeposit  Obtaininvest  \
count  1000.000000   1000.000000  968.000000     1000.000000    968.000000   
mean      0.782000    596.064330    0.054752        0.191000      0.026860   
std       0.413094   2328.221711    0.227613        0.393286      0.161756   
min       0.000000      2.000000    0.000000        0.000000      0.000000   
25%       1.000000    100.000000    0.000000        0.000000      0.000000   
50%       1.000000    300.000000    0.000000        0.000000      0.000000   
75%       1.000000    500.000000    0.000000        0.000000      0.000000   
max       1.000000  50000.000000    1.000000        1.000000      1.000000   

           Autobid  Transright  Riskdeposit  Thirdguarantee  
count  1000.000000  1000.00000   968.000000      968.000000  
mean      0.244000     0.17700     0.021694        0.034091  
std       0.429708     0.38186     0.145758        0.181557  
min       0.000000     0.00000     0.000000        0.000000  
25%       0.000000     0.00000     0.000000        0.000000  
50%       0.000000     0.00000     0.000000        0.000000  
75%       0.000000     0.00000     0.000000        0.000000  
max       1.000000     1.00000     1.000000        1.000000

In [47]:
q4_Data = q4_Data[q4_Data['Joinasso'].notnull()]
q4_Data.reset_index(drop = True, inplace = True)
q4_Data = q4_Data[q4_Data['Obtaininvest'].notnull()]
q4_Data.reset_index(drop = True, inplace = True)
q4_Data = q4_Data[q4_Data['Riskdeposit'].notnull()]
q4_Data.reset_index(drop = True, inplace = True)
q4_Data = q4_Data[q4_Data['Thirdguarantee'].notnull()]
q4_Data.reset_index(drop = True, inplace = True)

In [48]:
q4_Data.describe()

Collapse    RegCapital    Joinasso  Capitaldeposit  Obtaininvest  \
count  968.000000    968.000000  968.000000      968.000000    968.000000   
mean     0.799587    593.436568    0.054752        0.164256      0.026860   
std      0.400517   2365.067294    0.227613        0.370699      0.161756   
min      0.000000      2.000000    0.000000        0.000000      0.000000   
25%      1.000000    100.000000    0.000000        0.000000      0.000000   
50%      1.000000    300.000000    0.000000        0.000000      0.000000   
75%      1.000000    500.000000    0.000000        0.000000      0.000000   
max      1.000000  50000.000000    1.000000        1.000000      1.000000   

          Autobid  Transright  Riskdeposit  Thirdguarantee  
count  968.000000  968.000000   968.000000      968.000000  
mean     0.252066    0.182851     0.021694        0.034091  
std      0.434423    0.386744     0.145758        0.181557  
min      0.000000    0.000000     0.000000        0.000000  
25%      0.000000    0.000000     0.000000        0.000000  
50%      0.000000    0.000000     0.000000        0.000000  
75%      1.000000    0.000000     0.000000        0.000000  
max      1.000000    1.000000     1.000000        1.000000

In [49]:
# platform default (survival) likelihood : Collapse
q4_y = q4_Data[['Collapse']]
# platform characteristics 
q4_X = q4_Data[['RegCapital', 'Joinasso', 'Capitaldeposit', 'Obtaininvest', 'Autobid',
       'Transright', 'Riskdeposit', 'Thirdguarantee']]

### Perform the Cox model (Proportional hazards model)